group/assignment6/rearrange_cars.py

```diff
@@ -0,0 +1,64 @@
+import unittest
+
+def inverse_map(state):
+    """ Returns array indexed with previous values. """
+    inversed = len(state) * [0]
+
+    for position, car in enumerate(state):
+        inversed[car] = position
+
+    return inversed
+
+def move_car(start, start_position, current_empty):
+    """
+        Moves car from start_position to empty place.
+        Returns new empty place and configuration with moved car.
+    """
+    start[start_position], start[current_empty], current_empty = start[current_empty], start[start_position], start_position
+    return (current_empty, start)
+
+def find_sequence(start, end):
+    """
+        Yields sequence of moves between start and end states.
+    """
+    current_empty = start.index(0)
+    end_positions = inverse_map(end)
+    
+    for position in range(len(start)):
+        # At first, move car, that should be placed in current
```

Comment:
Thanks for the coherent comments, they really made this function easy to understand:)